compiler/souschef/src/Gaussian.cpp

```diff
@@ -50,22 +50,26 @@ std::vector<uint8_t> GaussianFloat32DataChef::generate(int32_t count) const
   return res;
 }
 
-std::vector<uint8_t> GaussianInt32DataChef::generate(int32_t count) const
+template <typename T> std::vector<uint8_t> GenerateGaussian(int32_t count, float mean, float stddev)
 {
   // TODO Support seed value override
   auto seed = std::chrono::system_clock::now().time_since_epoch().count();
 
   std::minstd_rand rand{static_cast<std::minstd_rand::result_type>(seed)};
-  std::normal_distribution<float> dist{_mean, _stddev};
+  std::normal_distribution<float> dist{mean, stddev};
 
   std::vector<uint8_t> res;
 
+  constexpr float max_cap = std::numeric_limits<T>::max();
+  constexpr float min_cap = std::numeric_limits<T>::min();
   for (uint32_t n = 0; n < count; ++n)
   {
-    auto const value = static_cast<int32_t>(dist(rand));
+    float raw_value = dist(rand);
+    const float capped_value = std::max(min_cap, std::min(max_cap, raw_value));
+    auto const value = static_cast<T>(capped_value);
```

Comment:
Or do you have concerns about this changes?

Reply:
Nope :)
code looks good as I imagine as new method intruduction of `template <typename T> std::vector<uint8_t> GenerateGaussian(int32_t count, float mean, float stddev)`

Reply:
Ok! I've created #6538

It differs from this commit: it actually unifies all generators, including floating point one.

Reply:
I want to merge this PR after 6538